src/modules/commander/Commander.cpp

```diff
@@ -327,169 +244,153 @@ int commander_main(int argc, char *argv[])
-				calib_ret = do_gyro_calibration(&mavlink_log_pub);
+				// accelerometer calibration: param5 = 1
+				send_vehicle_command(vehicle_command_s::VEHICLE_CMD_PREFLIGHT_CALIBRATION, 0.f, 0.f, 0.f, 0.f, 1.f, 0.f, 0.f);
 
 			} else if (!strcmp(argv[2], "level")) {
-				calib_ret = do_level_calibration(&mavlink_log_pub);
-
-			} else if (!strcmp(argv[2], "esc")) {
-				calib_ret = do_esc_calibration(&mavlink_log_pub, &armed);
+				// board level calibration: param5 = 1
+				send_vehicle_command(vehicle_command_s::VEHICLE_CMD_PREFLIGHT_CALIBRATION, 0.f, 0.f, 0.f, 0.f, 2.f, 0.f, 0.f);
 
 			} else if (!strcmp(argv[2], "airspeed")) {
-				calib_ret = do_airspeed_calibration(&mavlink_log_pub);
+				// airspeed calibration: param6 = 2
+				send_vehicle_command(vehicle_command_s::VEHICLE_CMD_PREFLIGHT_CALIBRATION, 0.f, 0.f, 0.f, 0.f, 0.f, 2.f, 0.f);
+
+			} else if (!strcmp(argv[2], "esc")) {
+				// ESC calibration: param7 = 1
+				send_vehicle_command(vehicle_command_s::VEHICLE_CMD_PREFLIGHT_CALIBRATION, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 1.f);
 
 			} else {
 				PX4_ERR("argument %s unsupported.", argv[2]);
-			}
-
-			if (calib_ret) {
-				PX4_ERR("calibration failed, exiting.");
 				return 1;
-
-			} else {
-				return 0;
 			}
 
+			return 0;
+
 		} else {
 			PX4_ERR("missing argument");
 		}
 	}
 
 	if (!strcmp(argv[1], "check")) {
-		bool preflight_check_res = Commander::preflight_check(true);
+		bool preflight_check_res = PreFlightCheck::preflightCheck(nullptr, status, status_flags, true, true, false, 30_s);
 		PX4_INFO("Preflight check: %s", preflight_check_res ? "OK" : "FAILED");
 
-		bool prearm_check_res = PreFlightCheck::preArmCheck(&mavlink_log_pub, status_flags, safety, arm_requirements);
+		bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, safety_s{},
+					PreFlightCheck::arm_requirements_t{});
 		PX4_INFO("Prearm check: %s", prearm_check_res ? "OK" : "FAILED");
 
 		return 0;
 	}
 
 	if (!strcmp(argv[1], "arm")) {
-		bool force_arming = false;
+		float param2 = 0.f;
 
+		// 21196: force arming/disarming (e.g. allow arming to override preflight checks and disarming in flight)
 		if (argc > 2 && !strcmp(argv[2], "-f")) {
-			force_arming = true;
+			param2 = 21196.f;
 		}
 
-		if (TRANSITION_CHANGED != arm_disarm(true, !force_arming, &mavlink_log_pub, "command line")) {
-			PX4_ERR("arming failed");
-		}
+		send_vehicle_command(vehicle_command_s::VEHICLE_CMD_COMPONENT_ARM_DISARM, 1.f, param2);
 
 		return 0;
 	}
 
 	if (!strcmp(argv[1], "disarm")) {
-		if (TRANSITION_DENIED == arm_disarm(false, true, &mavlink_log_pub, "command line")) {
-			PX4_ERR("rejected disarm");
-		}
+		send_vehicle_command(vehicle_command_s::VEHICLE_CMD_COMPONENT_ARM_DISARM, 0.f, 0.f);
 
 		return 0;
 	}
 
 	if (!strcmp(argv[1], "takeoff")) {
+		// switch to takeoff mode and arm
+		send_vehicle_command(vehicle_command_s::VEHICLE_CMD_NAV_TAKEOFF);
+		send_vehicle_command(vehicle_command_s::VEHICLE_CMD_COMPONENT_ARM_DISARM, 1.f, 0.f);
```

Comment:
Hang on, first you need to arm and then send a takeoff, right?

Reply:
That's indeed how it was done before: https://github.com/PX4/Firmware/blob/fbcc6a96bc89d14806126c14d70b9a0585dcdb6d/src/modules/commander/Commander.cpp#L412-L413
but I think QGC also switches to takeoff first and then arms.
If you'd ask me a takeoff command should do whatever is necassary to take off...